### PR TITLE
fix: make moneyInput button width consistent

### PR DESCRIPTION
**Pull request by mikknarusk**
## 🖼 Context
https://transferwise.atlassian.net/browse/SEND-1671

Example in product:
54e4e7df-18dd-4eec-97de-016635e32db2

Extreme example in dev:
Screenshot 2020-11-10 at 10 42 12
After (desktop):
Screenshot 2020-11-10 at 10 49 02
After (mobile):
Screenshot 2020-11-10 at 10 49 06

## 🚀 Changes
Adds a consistent `min-width` to the `MoneyInput` based on some tests.
 

## 🤔 Considerations
This is obviously not future proof, I'd love to hear some other ideas.

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/components/src/moneyInput/MoneyInput.story.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ export default {
 };
 
 export const basic = () => {
-  const size = select('size', Object.values(MoneyInput.Size), MoneyInput.Size.MEDIUM);
+  const size = select('size', Object.values(MoneyInput.Size), MoneyInput.Size.LARGE);
```
Comment (mikknarusk) [inline packages/components/src/moneyInput/MoneyInput.story.js:12]: Changed the default size here as that's what we use in the product, I haven't seen any other sizes in use anywhere.

File: packages/components/src/moneyInput/MoneyInput.less [hunk 2]
```diff
@@ -32,6 +32,15 @@
 .tw-money-input .amount-currency-select-btn > .tw-select > .btn {
   display: flex;
   align-items: center;
+  justify-content: space-between;
+
```
Comment (oliverviljamaa) [inline packages/components/src/moneyInput/MoneyInput.less:36]:
```suggestion
  justify-content: space-between;
```

**Comment by mikknarusk** [comment]
If I use `rem` instead, it works out of the box for all sizes, but it will looks increasingly bad in smaller sizes. But compared to the starting point I'd say it's fine:

LG:
Screenshot 2020-11-10 at 11 08 33

MD:
Screenshot 2020-11-10 at 11 08 36

SM:
Screenshot 2020-11-10 at 11 08 43

SM without my changes for reference:
Screenshot 2020-11-10 at 11 08 54

**Review by oliverviljamaa — APPROVED** [review]

File: packages/components/src/moneyInput/MoneyInput.story.js (hunk 1 above)
Comment (danlaush) [reply packages/components/src/moneyInput/MoneyInput.story.js:12]: It's the storybook entry so happy for this to change, but if you're curious I'm using Medium ([code](https://github.com/transferwise/lienzo-frontend/blob/master/components/PaypalCalculator/PaypalCalculator.js#L187), [url](https://transferwise.com/gb/paypal-fee-calculator/)). Also now realise I should be using an enum for the size 😅

**Review by fergusjordantw — APPROVED** [review]

File: packages/components/src/moneyInput/MoneyInput.story.js (hunk 1 above)
Comment (mikknarusk) [reply packages/components/src/moneyInput/MoneyInput.story.js:12]: Cool, happy to undo

**Comment by danlaush** [comment]
@mikknarusk did you consider the `ch` unit? https://meyerweb.com/eric/thoughts/2018/06/28/what-is-the-css-ch-unit/

**Comment by mikknarusk** [comment]
> @mikknarusk did you consider the `ch` unit? https://meyerweb.com/eric/thoughts/2018/06/28/what-is-the-css-ch-unit/
> I would guess that setting it to `3ch` wide would make it always big enough to fit `000` for the given font, so it should always have enough space for any 3-letter currency.

I didn't, interesting read. 
Looking at my change, I need to target the entire button, not just the three letters. Not because I want to, but because there isn't anything to target for just the text. The fact that the text itself is not a separate element + the fact that the span doesn't have a class + the fact that there's a second span makes it a bit inconvenient and flaky to target

File: packages/components/src/moneyInput/MoneyInput.less [hunk 3]
```diff
@@ -32,6 +32,13 @@
 .tw-money-input .amount-currency-select-btn > .tw-select > .btn {
   display: flex;
   align-items: center;
+
+  span:not(.tw-icon) {
+    min-width: 2.6em;
+    @media (--screen-sm) {
+      min-width: calc(~"2.6em +" + (var(--size-8) * 4));
```
Comment (helendurranttw) [inline packages/components/src/moneyInput/MoneyInput.less:39]: Unfortunately this won't work in IE11 - the custom props polyfill has trouble working with custom props inside calc statements. You'll need to remove the custom prop and swap it out for a normal value :)

**Review by danlaush — APPROVED** [review]

File: packages/components/src/moneyInput/MoneyInput.less (hunk 3 above)
Comment (mikknarusk) [reply packages/components/src/moneyInput/MoneyInput.less:39]: it seems to work fine locally, is it different in the prod build? 
Screenshot 2020-11-10 at 12 17 56

Comment (andreapiras) [reply packages/components/src/moneyInput/MoneyInput.less:39]: Why not using `--size-32` if it works anyway?

Comment (mikknarusk) [reply packages/components/src/moneyInput/MoneyInput.less]: @andreapiras I don't know, I like to keep things complicated 😂

Comment (helendurranttw) [reply packages/components/src/moneyInput/MoneyInput.less]: Oh! I'm happily mistaken. Though the build seems to be failing - did you notice? Maybe it's a different issue.

File: packages/components/src/moneyInput/MoneyInput.less [hunk 4]
```diff
@@ -32,6 +32,13 @@
 .tw-money-input .amount-currency-select-btn > .tw-select > .btn {
   display: flex;
   align-items: center;
+
+  span:not(.tw-icon) {
+    min-width: 2.6em; // fits a 3 letter string, ZMW being the longest one
+    @media (--screen-sm) {
+      min-width: calc(~"2.6em" + var(--size-32));
```
Comment (andreapiras) [inline packages/components/src/moneyInput/MoneyInput.less:39]: 😌

File: packages/components/src/moneyInput/MoneyInput.less (hunk 3 above)
Comment (mikknarusk) [reply packages/components/src/moneyInput/MoneyInput.less]: @helendurranttw it was this one https://github.com/transferwise/neptune-web/pull/666/commits/7e30eb9367fb8999250fce2004d8e9c13d447fbb

Comment (helendurranttw) [reply packages/components/src/moneyInput/MoneyInput.less]: Ah cool, thanks for that. And thanks for the contribution! 😁

**Review by helendurranttw — APPROVED** [review]

**Pull request merged**